src/email_service/health_check/lambda_function.py

```diff
@@ -43,9 +45,17 @@ def lambda_handler(event: dict, context) -> dict:
     formatted_service_name = format_service_name(SERVICE)
     logger.info("Health check started for service: %s", formatted_service_name)
 
+    repo = RunRepository()
+    if repo.check_connection():
+        status = "HEALTHY"
+        http_code = 200
+    else:
+        status = "UNHEALTHY"
+        http_code = 500
+
     # Build health check response
     health_status = {
-        "status": "HEALTHY",
+        "status": status,
         "service": formatted_service_name,
         "environment": ENVIRONMENT,
         "checked_at": datetime.datetime.now(datetime.UTC).strftime(TIME_FORMAT),
```

Comment:
I think we can provide more information about the health check API. That way, when we see a response of UNHEALTHY, the client would know which components are making the microservice unhealthy.